plugins/fuse/server/__init__.py

```diff
@@ -0,0 +1,67 @@
+import cherrypy
+
+from girder import logger
+from girder.models.file import File
+from girder.utility import config, mkdir
+
+from . import server_fuse
+
+
+MAIN_FUSE_KEY = '_server'
+
+
+def startFromConfig():
+    """
+    Check if the config file has a section [server_fuse] and key "path".  If
+    so, mount a FUSE at the specified path without using access validation.
+
+    :returns: True if a mount was made.  False if an error was raised.  None
+        if no mount was attemped.
+    """
+    cfg = config.getConfig().get('server_fuse', {})
+    path = cfg.get('path')
+    cherrypy.engine.subscribe('stop', server_fuse.unmountAll)
+
+    if path:
+        try:
+            mkdir(path)
+            return server_fuse.mountServerFuse(MAIN_FUSE_KEY, path, force=True)
+        except Exception:
+            logger.exception('Can\'t mount resource fuse: %s' % path)
+            return False
+
+
+def getFilePath(self, file):
+    """
+    Given a file resource, return a path on the local file system.  For
+    assetstores that have a fullPath method, this returns the results of that
+    call.  Otherwise, it returns a path on the main mounted FUSE.
+
+    :param file: file resource document.
+    :returns: a path on the local file system.
+    """
+    adapter = self.getAssetstoreAdapter(file)
+    if callable(getattr(adapter, 'fullPath', None)):
+        return adapter.fullPath(file)
+    return server_fuse.getServerFusePath(MAIN_FUSE_KEY, 'file', file)
+
+
+def getFuseFilePath(self, file):
+    """
+    Given a file resource, return a path on the main FUSE file system.
+
+    :param file: file resource document.
+    :returns: a path on the local file system.
+    """
+    return server_fuse.getServerFusePath(MAIN_FUSE_KEY, 'file', file)
+
+
+# Add two methods to the file model.
+#  getFilePath returns a local file path, but not necessarily in the FUSE.
+#  getFuseFilePath returns a local file path within the main FUSE.
+File.getFilePath = getFilePath
```

Comment:
It looks like these are only being used in the tests?  is there another reason these are being added directly to the File model?

Reply:
Yes -- I expect that most Girder plugins that will rely on the Fuse plugin will not want to care about the implementation details, and will just want a file path when necessary.  This extends the File model with the two functions.  Then, in a plugin like large_image, we could write:

```
# We don't need this conditional, if we know the fuse plugin is enabled
if hasattr(fileModel, 'getFilePath'):
    return fileModel.getFilePath(fileDoc)
adapter = fileModel.getAssetstoreAdapter(fileDoc)
if hasattr(adapter, 'fullPath'):
    return adapter.fullPath(fileDoc)
```

And, if we needed access to a file that might have a companion file (for instance, an item that has `foo` and `foo.xml`, and we are using an external program to parse `foo` that expects the xml file), then you would use `getFuseFilePath` to ensure that the two files appear side-by-each.

Reply:
To summarize from in person conversation,  I believe the plan is to:
1. Remove getFilePath from here and introduce it as an API on the File model in Girder (a separate PR)
2. When/if that is implemented duck-punch FIle.getFilePath to include Fuse functionality (in a separate PR)
3. Use getFuseFilePath as a utility function, don't attach it to File Model here.

Is that your recollection @manthey ?